alphalens/utils.py

```diff
@@ -330,6 +342,201 @@ def print_table(table, name=None, fmt=None):
+
+    groupby : pd.Series - MultiIndex or dict
+        Either A MultiIndex Series indexed by date and asset,
+        containing the period wise group codes for each asset, or
+        a dict of asset to group mappings. If a dict is passed,
+        it is assumed that group mappings are unchanged for the
+        entire time period of the passed factor data.
+    binning_by_group : bool
+        If True, compute quantile buckets separately for each group.
+        This is useful when the factor values range vary considerably
+        across gorups so that it is wise to make the binning group relative.
+        You should probably enable this if the factor is intended
+        to be analyzed for a group neutral portfolio
+    quantiles : int or sequence[float]
+        Number of equal-sized quantile buckets to use in factor bucketing.
+        Alternately sequence of quantiles, allowing non-equal-sized buckets
+        e.g. [0, .10, .5, .90, 1.] or [.05, .5, .95]
+        Only one of 'quantiles' or 'bins' can be not-None
+    bins : int or sequence[float]
+        Number of equal-width (valuewise) bins to use in factor bucketing.
+        Alternately sequence of bin edges allowing for non-uniform bin width
+        e.g. [-4, -2, -0.5, 0, 10]
+        Chooses the buckets to be evenly spaced according to the values
+        themselves. Useful when the factor contains discrete values.
+        Only one of 'quantiles' or 'bins' can be not-None
+    groupby_labels : dict
+        A dictionary keyed by group code with values corresponding
+        to the display name for each group.
+    max_loss : float, optional
+        Maximum percentage (0.00 to 1.00) of factor data dropping allowed,
+        computed comparing the number of items in the input factor index and
+        the number of items in the output DataFrame index.
+        Factor data can be partially dropped due to being flawed itself
+        (e.g. NaNs), not having provided enough price data to compute
+        forward returns for all factor values, or because it is not possible
+        to perform binning.
+        Set max_loss=0 to avoid Exceptions suppression.
+
+    Returns
+    -------
+    merged_data : pd.DataFrame - MultiIndex
+        A MultiIndex Series indexed by date (level 0) and asset (level 1),
+        containing the values for a single alpha factor, forward returns for
+        each period, the factor quantile/bin that factor value belongs to, and
+        (optionally) the group the asset belongs to.
+
+        - forward returns column names follow  the format accepted by
+          pd.Timedelta (e.g. '1D', '30m', '3h15m', '1D1h', etc)
+
+        - 'date' index freq property (merged_data.index.levels[0].freq) will be
+          set to a trading calendar (pandas DateOffset) inferred from the input
+          data (see infer_trading_calendar for more details). This is currently
+          used only in cumulative returns computation
+        ::
```

Comment:
This is not correct because 'date' index freq property is not inferred but it is copied from forward_returns